packages/SwingSet/src/kernel/vat-warehouse.js

```diff
@@ -168,14 +168,26 @@ export function makeVatWarehouse(kernelKeeper, vatLoader, policyOptions) {
     // instantiate all static vats
     for (const [name, vatID] of kernelKeeper.getStaticVats()) {
       logStartup(`provideVatKeeper for vat ${name} as vat ${vatID}`);
-      // eslint-disable-next-line no-await-in-loop
+      // Nested await safe because "terminal-combined-control-flow".
+      //
+      // `ensureVatOnline` is an async function, so this turn boundary always
+      // happens at the end of each iteration. However, the control-flow is
+      // still unbalanced. If the loop iterates zero times, we proceed to
+      // the next code synchronously. Else we proceed asynchronously.
+      // However, the next loop is terminal, and each iteration of that loop
+      // resemble each iteration of this one. Considered together, in all
+      // cases the first iteration of one of these loops will happen first,
+      // if there are any. And if there are more iterations from either loop,
+      // a turn boundary separated each iteration from the next.
+      // eslint-disable-next-line no-await-in-loop, @jessie.js/no-nested-await
```

Comment:
I concur with the justifications in this file, but yes let's do refer to the full explanations by reference rather than having them inline.

Reply:
e.g. https://gist.github.com/turadg/e7d7645789218173a578508ae7640365#file-terminal-combined-control-flow